tests/integration/conftest.py

```diff
@@ -286,22 +286,48 @@ def authenticated_member_client(
     factories.user_role.create(user_id=user.id, user=user, role=RoleEnum.MEMBER, grant=grant)
 
     login_user(user)
+    anonymous_client.user = user
+    anonymous_client.grant = grant
+    db_session.commit()
+
+    yield anonymous_client
+
+
+# TODO: combine (at least) the grant clients and allow the user+grant to come from another fixture so that we don't
+#       need to attach them to the client; tests that want access to the configured user+grant could request the
+#       fixture directly? Also maybe a pytest mark could be used to set the role provided for the authenticated grant
```

Comment:
Agree this feels like a nice explicit interface where its clear what you're asking for and getting